Sources/Metrics/Metrics.swift

```diff
@@ -98,5 +99,41 @@ extension Timer {
 
         self.recordNanoseconds(nanoseconds.partialValue)
     }
+
+    /// Convenience for measuring duration of a closure.
+    ///
+    /// - Parameters:
+    ///    - clock: The clock used for measuring the duration. Defaults to the continuous clock.
+    ///    - body: The closure to record the duration of.
+    @inlinable
+    @available(macOS 13, iOS 16, tvOS 16, watchOS 9, *)
+    public func measure<Result, Clock: _Concurrency.Clock>(
+        clock: Clock = .continuous,
+        body: () throws -> Result
+    ) rethrows -> Result where Clock.Duration == Duration {
+        let start = clock.now
+        defer {
+            self.record(start.duration(to: clock.now))
+        }
+        return try body()
+    }
+
+    /// Convenience for measuring duration of a closure with a provided clock.
```

Comment:
Nit: the doc for this method and the non-async version don't match. Maybe remove the last bit to match that one, since we're defaulting the `clock` param anyways?
```suggestion
    /// Convenience for measuring duration of a closure.
```